crates/bevy_ecs/src/query/state.rs

```diff
@@ -172,6 +240,85 @@ where
         }
     }
 
+    /// Returns the query results for the given array of [`Entity`].
+    ///
+    /// In case of a nonexisting entity or mismatched component, a [`QueryEntityError`] is
+    /// returned instead.
+    ///
+    /// ```rust
+    /// use bevy_ecs::prelude::*;
+    /// use bevy_ecs::query::QueryEntityError;
+    ///
+    /// #[derive(Component, PartialEq, Debug)]
+    /// struct A(usize);
+    ///
+    /// let mut world = World::new();
+    /// let mut entities: [Entity; 3] = [Entity::from_raw(0); 3];
+    ///
+    /// world.spawn().insert(A(42));
+    ///
+    /// for i in 0..3 {
+    ///     entities[i] = world.spawn().insert(A(i)).id();
+    /// }
+    ///
+    /// world.spawn().insert(A(73));
+    ///
+    /// let mut query_state = world.query::<&mut A>();
+    ///
+    /// let mut mutable_component_values = query_state.get_multiple_mut(&mut world, entities).unwrap();
+    ///
+    /// for mut a in mutable_component_values.iter_mut(){
+    ///     a.0 += 5;
+    /// }
+    ///
+    /// let component_values = query_state.get_multiple(&world, entities).unwrap();
+    ///
+    /// assert_eq!(component_values, [&A(5), &A(6), &A(7)]);
+    ///
+    /// let wrong_entity = Entity::from_raw(365);
+    ///
+    /// assert_eq!(query_state.get_multiple_mut(&mut world, [wrong_entity]), Err(QueryEntityError::NoSuchEntity(wrong_entity)));
```

Comment:
This line, and the line below are failing with 

```
error[E0369]: binary operation `==` cannot be applied to type `Result<[Mut<'_, A>; 2], QueryEntityError>`
```
This seems to be because Mut doesn't have an `Eq` blanket impl. I really really think it should, but I'm nervous to add it to this PR.

Reply:
You can use `.get_multiple_mut(..).unwrap_err()`